src/sinks/util/tcp.rs

```diff
@@ -211,17 +206,18 @@ impl Sink for TcpSink {
                 // This can return a proper disconnect error or `Ok(0)`
                 // which means the pipe is broken and we should try to reconnect.
                 //
-                // If this returns `WouldBlock` we know the connection is still
+                // If this returns `Poll::Pending` we know the connection is still
                 // valid and the write will most likely succeed.
-                let stream: &mut MaybeTlsStream<TcpStream> =
-                    connection.get_mut().get_mut().get_mut().get_mut();
-                match stream.read(&mut [0u8; 1]) {
-                    Err(error) if error.kind() != ErrorKind::WouldBlock => {
+
+                let stream: &mut MaybeTlsStream<TcpStream> = connection.get_mut().get_mut();
+                let mut cx = Context::from_waker(&noop_waker_ref());
+                match Pin::new(stream).poll_read(&mut cx, &mut [0u8; 1]) {
```

Comment:
Again, why the change here? This doesn't seem directly related to upgrading tokio.